pandas/tseries/index.py

```diff
@@ -64,6 +64,7 @@ def f(self):
             if self.tz is not utc:
                 values = self._local_timestamps()
 
+        # boolean accessors -> return array
```

Comment:
Because the handling of NaNs is different (that is related to the other issue of NaT not having the boolean fields, will open an issue about that). For `is_leap_year` (which returns False for NaT), the handling of missing values in `self._maybe_mask_results(result, convert='float64')` would return the wrong result.